api/utils/aws/endpoint.go

```diff
@@ -93,6 +93,12 @@ type RDSEndpointDetails struct {
 	ProxyCustomEndpointName string
 	// Region is the AWS region the database resides in.
 	Region string
+	// EndpointType specifies the type of the endpoint, if available.
+	//
+	// Note that the endpoint type of RDS Proxies are determined by their
+	// targets, so the endpoint type will be empty for RDS Proxies here as it
+	// cannot be decided by the endpoint URL itself.
```

Comment:
I don't have the context to know whether this is worthwhile or possible, but worth considering whether this is worthy of its own `RDSProxyUndecided` value to distinguish it from other reasons the field might be empty.

Reply:
It's a good point. First, it's not necessary since the endpoint type for RDS proxy is not required for this change. Second, usually (at least in the current impl), the empty string indicates unknown endpoint type:
https://github.com/gravitational/teleport/blob/f1a49168e24d89b422fb6517434d09a27d23e328/api/types/database.go#L1025-L1038

So I will leave it as it is for now.

Thanks